OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventService.cs

```diff
@@ -86,20 +107,93 @@ public async Task Delete(Guid id)
     {
         logger.LogTrace($"Deleting CompetitiveEvent with Id = {id} started.");
 
-        var entity = new CompetitiveEvent() { Id = id };
+        var entity = await competitiveEventRepository.GetById(id);
 
         try
         {
             await competitiveEventRepository.Delete(entity).ConfigureAwait(false);
 
             logger.LogTrace($"CompetitiveEvent with Id = {id} succesfully deleted.");
         }
-        catch (DbUpdateConcurrencyException)
+        catch (Exception) // DbUpdateConcurrencyException
         {
             logger.LogError("Deleting failed. CompetitiveEvent with Id = {Id} doesn't exist in the system", id);
             throw new ArgumentOutOfRangeException(
                 nameof(id),
                 localizer[$"CompetitiveEvent with Id = {id} doesn't exist in the system"]);
         }
     }
+
+    private async Task ChangeJudges(CompetitiveEvent currentCompetitiveEvent, List<JudgeDto> judgeDtoList)
+    {
+        var deletedIds = currentCompetitiveEvent.Judges
+            .Select(x => x.Id)
+            .Except(judgeDtoList.Select(x => x.Id))
+            .ToList();
+
+        if (deletedIds.Count > 0)
+        {
+            var judgeToDelete = currentCompetitiveEvent.Judges
+                .Where(judge => deletedIds.Contains(judge.Id))
+                .ToList();
+
+            var deleteTasks = judgeToDelete
+                .Select(deletedJudge => judgeRepository.Delete(deletedJudge));
```

Comment:
What is the maximum number of the parallel connection that we're allowed to open to the database ?

Reply:
Currently, this is a placeholder as the real logic is not yet defined. However, objectively, we expect there will be no more than 10 parallel connections in the worst case.